edgelet/iotedged/src/lib.rs

```diff
@@ -712,6 +759,129 @@ fn manual_provision(
     tokio_runtime.block_on(provision)
 }
 
+fn external_provision_payload(key: &str) -> (DerivedKeyStore<MemoryKey>, MemoryKey) {
+    let memory_key = MemoryKey::new(key);
+    let mut memory_hsm = MemoryKeyStore::new();
+    memory_hsm.insert(&KeyIdentity::Device, "primary", memory_key.clone());
+
+    let derived_key_store = DerivedKeyStore::new(memory_key.clone());
+    (derived_key_store, memory_key)
+}
+
+fn external_provision_tpm(
+    hsm_lock: Arc<HsmLock>,
+) -> Result<(DerivedKeyStore<TpmKey>, TpmKey), Error> {
+    let tpm = Tpm::new().context(ErrorKind::Initialize(
+        InitializeErrorReason::ExternalProvisioningClient,
+    ))?;
+
+    let tpm_hsm = TpmKeyStore::from_hsm(tpm, hsm_lock).context(ErrorKind::Initialize(
+        InitializeErrorReason::ExternalProvisioningClient,
+    ))?;
+
+    tpm_hsm
+        .get(&KeyIdentity::Device, "primary")
+        .map_err(|err| {
+            Error::from(err.context(ErrorKind::Initialize(
+                InitializeErrorReason::ExternalProvisioningClient,
+            )))
+        })
+        .and_then(|k| {
+            let derived_key_store = DerivedKeyStore::new(k.clone());
+            Ok((derived_key_store, k))
+        })
+}
+
+//fn external_provision<K>(
```

Comment:
Remove this commented-out code?

Reply:
ah shoot, forgot to clean this up. thanks!